### ISSUE TITLE
Restoring from backup fails silently on Android 11 with certain file managers

### ISSUE THREAD

**Issue by DenizenB** [signalapp/Signal-Android#10387]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

Restoring from a backup on Android 11 may fail depending on which file manager is used to browse for the backup file.

I'm not sure if this is the fault of Signal or the file managers, but in either case,  I'm submitting this issue in case others are facing the same problem.

**TL;DR:** If you're facing the same issue as described below, use Simple File Manager (available on F-droid) to browse for your backup file.

### Steps to reproduce

- Open clean install of Signal
- Press `Restore backup`
- Press `Choose backup`
- Select backup file with your file manager of choice

**Actual result:** Signal silently skips to the "Enter phone number" screen.
**Expected result:** Signal should confirm the size of my backup and prompt me for my passphrase.

### Additional notes

The above behavior has been observed with the following file managers:
- OnePlus File Manager (com.oneplus.filemanager)
- Xiaomi File Manager (com.mi.android.globalFileexplorer)

In addition to this, *Files by Google* doesn't even show up as an option in the `Complete action using...` dialog.

I was able to successfully restore from a backup using Simple File Manager (available on F-droid).

### Screenshots

![output](https://user-images.githubusercontent.com/6432036/104075879-31970c80-520c-11eb-8e91-29282b5f0c67.gif)

### Device info

**Device:** OnePlus 8T
**Android version:** 11.0.5.6.KB05BA
**Signal version:** 5.1.1 (playProd debug)

### Link to debug log

Highlighted log messages:
```
2021-01-08 23:41:18.148 11389-5208/org.thoughtcrime.securesms W/BackupUtil: isBackupFileReadable: The document at the specified Uri cannot be read.
2021-01-08 23:41:18.148 11389-5208/org.thoughtcrime.securesms W/BackupUtil: Could not load backup info.
...
2021-01-08 23:41:18.165 11389-11389/org.thoughtcrime.securesms I/RestoreBackupFragment: Skipping backup detection. No backup found, or permission revoked since.
```

Full logcat output: https://pastebin.com/WkBrR9K9

**Comment by androdebugur** [comment]
The current version of Signal app on play store is 5.0.3, how do you have 5.1.1. What does "playProd debug" mean?

**Comment by UserX404** [comment]
@androdebugur 
It's a beta and not a production version.
You are on beta release channel.

**Comment by RobertBuhren** [comment]
I have the same problem with my Pixel 5. However, the propose workaround does not work for me, even tough I installed the Simple File Manager, there is no possibility to select the file manager when trying to open the backup file.

I'm using Signal 5.1.9

Regards,

Robert

**Comment by jumoog** [comment]
I have the same Problem with OnePlus 8T and Android 11, Is there a workarround?

**Comment by jumoog** [comment]
I think thats related to Storage changes in Android 11 https://developer.android.com/about/versions/11/privacy/storage

**Comment by jumoog** [comment]
It is! If you put the Backup into the `Download` Folder Signal can access it!

**Comment by sjurtf** [comment]
I have this exact same issue. Workaround described in $10918 does not work for me.

Signal 5.3.12
Google Pixel 3
Android 11, Security Patch Feb 2021

**Comment by robert7k** [comment]
I can confirm this bug when using [Amaze File Manager](https://play.google.com/store/apps/details?id=com.amaze.filemanager&hl=de&gl=US) on LineageOS 18.1 (Android 11, Patch Level March 2021, Signal 5.6.4)

Using the stock file manager fixed the problem.
